mitodl/mit-open
# Learning format filter for search/db api's

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4096

### Description (What does it do?)
Renames `LearningResource.format` to `learning_format`, adds a filter parameter of the same name ("format" is already reserved as a DRF parameter: `format=json`, for example), and modifies how the field is serialized.


### How can this be tested?
Run the following:
```
./manage.py recreate_index --all
./manage.py backpopulate_prolearn_data
./manage.py backpopulate_xpro_data
./manage.py backpopulate_mitxonline_data
```

Then try out the following urls:

http://localhost:8063/api/v1/learning_resources/?learning_format=hybrid should return about 13 resources
http://localhost:8063/api/v1/learning_resources/?learning_format=in_person should return about 96 resources
http://localhost:8063/api/v1/learning_resources/?learning_format=online should return about lots of resources (potentially thousands if you imported all OCW courses previously).

Repeat the above urls with opensearch instead, you should get the same numbers:
http://localhost:8063/api/v1/learning_resources_search/?learning_format=hybrid
http://localhost:8063/api/v1/learning_resources_search/?learning_format=in_person 
http://localhost:8063/api/v1/learning_resources_search/?learning_format=online
